src/components/templates/PosterHall/components/PosterPreview/PosterPreview.tsx

```diff
@@ -39,13 +74,78 @@ export const PosterPreview: React.FC<PosterPreviewProps> = ({
     [categories]
   );
 
+  const { relatedVenueIds } = useRelatedVenues({
+    //isLoading, relatedVenues,
+    currentVenueId: venueId,
+  });
+
+  const {
+    // isEventsLoading,
+    events: relatedVenueEvents = emptyRelatedEvents,
+  } = useVenueEvents({
+    venueIds: relatedVenueIds,
+  });
+
+  const bookmarkPoster = useCallback(() => {
+    setBookmarkPoster(!isBookmarkedPoster);
+    personalizedPoster.isSaved = !personalizedPoster.isSaved;
+    if (userId && personalizedPoster.id) {
+      savePosterToProfile({
+        venueId: personalizedPoster.id,
+        userId: userId,
+        removeMode: !personalizedPoster.isSaved,
+      });
+    }
+    relatedVenueEvents
+      .filter((event) => event.venueId === personalizedPoster.id)
+      .map((event) => {
+        userId &&
+          event.id &&
+          updatePersonalizedSchedule({
+            event: event,
+            userId: userId,
+            removeMode: !personalizedPoster.isSaved,
+          });
+        return {};
+      });
+    return relatedVenueEvents;
+  }, [userId, isBookmarkedPoster, personalizedPoster, relatedVenueEvents]);
+
+  const onBookmarkPoster: MouseEventHandler<HTMLDivElement> = useCallback(
+    (e) => {
+      e.stopPropagation();
+      bookmarkPoster();
+    },
+    [bookmarkPoster]
+  );
```

Comment:
We don't want to use this 'passive' `onX` naming style for functions that 'do things'. They should be named for the action they do.

In this case `bookmarkPoster` would be the most obvious name, but it's already been used.

I can't see any reason why we need 2 separate `useCallback`'d functions here, so they can likely be merged into a single definition, called `bookmarkPoster`.

Reply:
Also, why do we need `e.stopPropagation();` here? This is something we've been trying to eliminate from our code recently.

Reply:
Any tips on how to resolve this would be helpful. As noted on the navbar schedule issue, I can't seem to find a workaround that doesn't call both the bookmark and underlying poster link when clicked without the `e.stopPropagation()`